### PR TITLE
feat(streaming): implement  superfluid sdk and react hooks

### LINKED ISSUE #23: Superfluid Streaming SDK (Celo + Base)
## 1) Motivation

We want an SDK that uses G$ SuperToken capabilities and makes it easy for developers and dApps to offer streaming solutions. The repo currently lacks a Superfluid streaming SDK, so a future UI cannot reliably create or manage streams. Build a new SDK package that supports Celo for G$ and Base for SUP‑related activity, and combines live contract reads with Superfluid subgraph queries for historical and aggregate data.

Expected output and supported features:

- A new `@goodsdks/*` package that exposes a typed API for streaming.
- Stream lifecycle support: create, update (flow rate), delete.
- Fetch running streams (live).
- SuperToken holdings and balances via subgraph queries.
- Distribution pools (GDA) display via subgraph queries. (supporting connecting/disconnecting from GDA pools)
- Core SDK plus React hooks compatible SDK.
- Support G$ dev/staging/prod tokens (dev should be easily able to choose which env the SDK is initialized with)

## 2) Guidelines

- Use `@sfpro/sdk` and follow https://sdk.superfluid.pro
  Make sure to read up on the superfluid pro documentation.
  There are both hooks for react environments + core methods for non-react environments
  See:https://sdk.superfluid.pro/docs/architecture
- Enforce chain validation for Celo and Base.
- Prefer subgraph data fetching over direct contract reads when feasible.
- Retrieve Superfluid protocol addresses via `@sfpro/sdk` address maps (e.g. `cfaForwarderAddress[chainId]`, `hostAddress[chainId]`) and document this approach.
- Add new package scaffolding under `packages/<new-sdk>/` with `tsup` build and public exports from `src/index.ts`.
- Include basic usage docs and examples.

## 4) Starting points (top paths + notes)

- `packages/savings-sdk/src/viem-sdk.ts` — Celo-only SDK structure and viem read/write patterns.
- `packages/engagement-sdk/src/viem-sdk.ts` — SDK shape and log/event fetching conventions.
- `packages/engagement-sdk/src/utils/rpc.ts` — block batching helpers for RPC log scans.
- `packages/react-hooks` — wagmi hook pattern (create a new streaming-hooks following the structure of react-hooks).
- `packages/citizen-sdk/src/constants.ts` — chain config and Celo address patterns.
- `packages/citizen-sdk/src/utils/chains.ts` — chain/env validation helpers.
- https://docs.superfluid.org/docs/sdk/money-streaming/subgraph — Superfluid subgraph docs.
- https://explorer.superfluid.org/subgraph — Superfluid subgraph playground (select chain tab).
- https://sdk.superfluid.pro/ — It supports LLM-friendly SDK notes through https://sdk.superfluid.pro/llm.txt.
  - Note: use subgraph queries to surface GDA distribution pools (pool id, members, connection status).

Reference contracts:

- https://github.com/GoodDollar/GoodProtocol/blob/master/contracts/token/superfluid/SuperToken.sol
- https://github.com/GoodDollar/GoodProtocol/blob/master/contracts/token/superfluid/SuperGoodDollar.sol

Sample subgraph queries:

```graphql
# Fetch token snapshot balances
query ($id: ID = "") {
  account(id: $id) {
    accountTokenSnapshots {
      balanceUntilUpdatedAt
      token {
        id
        name
      }
      updatedAtTimestamp
    }
  }
}
```

```graphql
# See pool-membership
query ($id: ID = "") {
  account(id: $id) {
    poolMemberships(where: { account: $id }) {
      isConnected
      pool {
        id
      }
    }
  }
}
```

```graphql
# Distribution pools (GDA) - example pool details
query ($id: ID = "") {
  pool(id: $id) {
    id
    totalUnits
    totalAmountClaimed
    flowRate
  }
}
```

SUP‑specific subgraph (for SUP reserve holdings):

- https://thegraph.com/explorer/subgraphs/6dRuPxMvaJAp32hvcTsYbAya69A4t1KUHh2EnV3YQeXU?view=Query&chain=arbitrum-one

```graphql
# Reserve addresses for SUP holdings
{
  lockers(where: { lockerOwner: "0x0" }) {
    id
    lockerOwner
    blockNumber
    blockTimestamp
  }
}
```

## 5) Definition of Done (DoD) + How to test

**DoD**

- [ ] New package exists under `packages/<new-sdk>/`
- [ ] Public exports are surfaced via `packages/<new-sdk>/src/index.ts`.
- [ ] Typed methods exist for stream create/update/delete, list running streams.
- [ ] Uses `@sfpro/sdk` with Celo + Base validation.
- [ ] Superfluid addresses are pulled via `@sfpro/sdk` address maps and documented.
- [ ] Its using the right G$ addresses, based on env (has to support both development and production)
- [ ] Subgraph integration supports historical balances/holdings and pool membership queries.
- [ ] Distribution pools (GDA) can be listed and displayed (pool details + membership status) via subgraph queries.
- [ ] It should be possible to connect/disconnect active GDA streams
- [ ] SUP reserve holdings can be queried via the SUP‑specific subgraph endpoint.
- [ ] Listing streams uses safe batching/limits to avoid unbounded RPC or subgraph scans.
- [ ] Basic usage docs include CRUD + listing + subgraph examples.
- [ ] Build passes for the new workspace.

### PR DESCRIPTION
**Pull request by HushLuxe**
I'm adding the @goodsdks/streaming-sdk and @goodsdks/react-hooks (streaming module) to the monorepo. This provides a unified, type-safe way to handle Superfluid operations on Celo and Base, specifically optimized for G$ and GDA pools.

### Changes
- Implemented core StreamingSDK and GdaSDK in packages/streaming-sdk.
- Added React hooks for stream management and GDA pool connectivity in packages/react-hooks.
- Centralized protocol addresses and subgraph URLs in a single constants file.
- Added comprehensive documentation and a testing guide.

### Testing
I've verified the implementation with:
- Unit Tests: All 37 tests for the core SDK and utilities are passing.
- Build Verification: Both CommonJS and ESM builds are working as expected.
- UI Testing: Manually tested stream creation/deletion and GDA pool connections in the demo app.
- CI Prep: Ran turbo build and turbo lint across the workspace to ensure no regressions.

### Checklist:
- [x] PR title follows convention: [(Feature) Superfluid Streaming SDK Implementation]
- [x] Code follows project style guidelines
- [x] Unit tests pass locally and cover the new functionality

## Summary by Sourcery

Add a new Superfluid-based streaming SDK for Celo/Base and expose React hooks for managing streams and GDA pool interactions.

New Features:
- Introduce @goodsdks/streaming-sdk providing StreamingSDK, GdaSDK, subgraph client, and utilities for Superfluid streams, GDA pools, and SUP reserve queries across supported chains and environments.
- Add React streaming hooks in @goodsdks/react-hooks for listing streams, querying GDA pools and memberships, managing SUP reserves, and mutating streams and pool memberships via React Query.

Enhancements:
- Centralize Superfluid protocol addresses, G$ SuperToken addresses, subgraph URLs, and chain metadata in shared constants for streaming operations.
- Provide comprehensive README and TESTING documentation for the streaming SDK, including usage examples, configuration, and validation steps.

Tests:
- Add a Vitest test suite for the streaming SDK covering utilities, chain validation, core StreamingSDK and GdaSDK behaviors, error handling, and edge cases.

### CODE REVIEW

**Review by sirpy — DISMISSED** [review]

File: apps/engagement-app/.vercelignore [hunk 1]
```diff
@@ -0,0 +1,10 @@
+node_modules
```
[the remark this answers is not in the document]
Comment (sirpy) [reply apps/engagement-app/.vercelignore]: also another option is to upgrade vercel version

File: apps/demo-identity-app/.env.example [hunk 2]
```diff
@@ -0,0 +1 @@
+VITE_GRAPH_API_KEY=
```
[the remark this answers is not in the document]
Comment (HushLuxe) [reply apps/demo-identity-app/.env.example]: I’ve added a frontend for the demo

File: apps/demo-streaming-app/src/App.tsx [hunk 3]
```diff
@@ -0,0 +1,535 @@
...
+        data: streams,
+        isLoading: streamsLoading,
+        refetch: refetchStreams,
+    } = useStreamList({
+        account: address as Address,
+        environment,
+        enabled: !!address,
+    }) as { data: StreamInfo[] | undefined, isLoading: boolean, refetch: () => void }
+
+    const { data: pools, isLoading: poolsLoading } = useGDAPools({
```
Comment (L03TJ3) [inline apps/demo-streaming-app/src/App.tsx]: You are using here an environment param but its not defined as being a param for this hook?

File: packages/streaming-sdk/README.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+  receiver: '0x...',
+  token?: 'G$' | 'SUP' | Address,
+  onHash?: (hash) => console.log(hash)
+})
+```
+
+#### `getActiveStreams(account, direction?)`
+Returns active streams for an account.
+```typescript
+const streams = await sdk.getActiveStreams('0x...', 'outgoing' | 'incoming' | 'all')
```
Comment (L03TJ3) [inline packages/streaming-sdk/README.md]: Docs seem out of sync. the implementation expects an options object.
in this example, '0x...' is treated as options and the rest will be ignored

**Review by L03TJ3 — CHANGES_REQUESTED** [review]

**Review by L03TJ3 — CHANGES_REQUESTED** [review]

**Review by L03TJ3 — CHANGES_REQUESTED** [review]

**Review by L03TJ3 — CHANGES_REQUESTED** [review]

**Comment by L03TJ3** [comment]
Hey, making sure you saw my latest comments @HushLuxe

**Comment by HushLuxe** [comment]
Thanks for the review@L03TJ3 
Already working on it

**Comment by HushLuxe** [comment]
Hi @L03TJ3... I have made an adjustment

The SDK README now includes `getBalanceHistory`, address resolution, SUP reserve handling, and `getActiveStreams()` pagination. I also fixed the pagination behavior so `{ first: 20 }` works correctly without requiring `skip`, and added live flow reads via `getFlowRate` / `getFlowInfo`.

On the hooks/demo side, I added `useStreamingSDK`, `useSuperTokenBalance`, and `useBalanceHistory`, exposed `first` / `skip` on `useStreamList`, and updated the demo to show token balances and balance history alongside the existing stream and GDA views.

I also re-ran:
- `yarn workspace @goodsdks/streaming-sdk test --run`
- `yarn workspace @goodsdks/react-hooks build`
- `yarn workspace demo-streaming-app check-types`

Happy to fix anything else quickly

File: apps/demo-streaming-app/src/App.tsx [hunk 5]
```diff
@@ -0,0 +1,859 @@
...
+        token: selectedToken,
+        environment,
+        enabled: !!address && !!RESOLVED_TOKEN_ADDR,
+    })
+
+    const {
+        data: balanceHistory,
+        isLoading: balanceHistoryLoading,
+        refetch: refetchBalanceHistory,
+    } = useBalanceHistory({
```
Comment (L03TJ3) [inline apps/demo-streaming-app/src/App.tsx:535]: there is no from/to timestamp supplied, and we don't seem to have support for a default range (maybe last 30 days)

So this demo will never show anyone balance (that is also true for sup balance)

**Review by L03TJ3 — CHANGES_REQUESTED** [review]

File: apps/demo-streaming-app/src/App.tsx (hunk 5 above)
Comment (HushLuxe) [reply apps/demo-streaming-app/src/App.tsx:535]: Alright @L03TJ3 
 Can I close this current pr and open another one?

Comment (L03TJ3) [reply apps/demo-streaming-app/src/App.tsx:535]: No, we need to have a pull-request where the history of discussion is clear.
re-opening new PR's makes it very hard to continue reviewing with having to cross-reference what was done/already reviewed commented on etc.

Comment (HushLuxe) [reply apps/demo-streaming-app/src/App.tsx:535]: I have updated the demo (`App.tsx`) to pass an explicit rolling 30-day window to `useBalanceHistory()`

Also:
- Updated the balance snapshot label to clearly say it’s showing the last 30 days.
- Improved the “Active Streams” section labels for clarity. The direction label now shows `From:` when the connected wallet is receiving, and `To:` when it’s sending

File: apps/demo-identity-app/vite.config.mts [hunk 6]
```diff
@@ -29,6 +29,22 @@ export default defineConfig({
   plugins: [react()],
   define: {
     "process.browser": true,
-    "process.env": process.env,
+    "process.env.NODE_ENV": JSON.stringify(process.env.NODE_ENV ?? "development"),
+  },
+  build: {
```
Comment (L03TJ3) [inline apps/demo-identity-app/vite.config.mts]: this seems like a hack or skipping an error? what is this about?

File: yarn.lock
Comment (L03TJ3) [inline yarn.lock:1]: The yarn lock file is broken.
Way to resolve is: copy over the existing yarn-lock from main branch.
this should not have any changes.
and run a clean yarn install against it.

I am not able to run yarn install or yarn build given this current yarn.lock.
Yarn.lock should never be manually fixed or adjusted.

**Review by L03TJ3 — CHANGES_REQUESTED** [review]

File: yarn.lock
Comment (HushLuxe) [reply yarn.lock:1]: I have replaced `yarn.lock` with the current `main` version and regenerated it with a clean `yarn install --mode=skip-build`

File: apps/demo-identity-app/vite.config.mts (hunk 6 above)
Comment (HushLuxe) [reply apps/demo-identity-app/vite.config.mts]: not sure where you saw this,but i did not work on this file, It probably came from an older commit in the PR history that was later reverted. That file is not part of the current streaming diff anymore.

Comment (HushLuxe) [reply apps/demo-identity-app/vite.config.mts]: Hi @L03TJ3 

I verified this file came from an older commit in the PR history and was later reverted weeks ago, so it is not part of the current streaming diff

File: yarn.lock
Comment (HushLuxe) [reply yarn.lock:1]: I also re-ran the branch checks after regenerating it:

- `yarn workspace @goodsdks/good-reserve build`
- `yarn workspace @goodsdks/react-hooks build`
- `yarn workspace @goodsdks/streaming-sdk test --run`
- `yarn workspace demo-streaming-app check-types`
- `yarn workspace demo-streaming-app build`
- `yarn build`

They  all passes with the regenerated lockfile

**Review by L03TJ3 — APPROVED** [review]
